### PR TITLE
Add extensions catalyst support

### PR DESCRIPTION
**Pull request by YUNQIUGUO**
The main motivation for this is ORT Nuget Package now (sort of) has a dependency of OnnxRuntime.ML.Extensions/Dummy package. We recently added ios-maccatalyst target in onnxruntime.xcframework for iOS target. Upon testing, .NET MAUI app will complain about the extensions nuget package ios native runtime not having supported platform framework for catalyst.

No other quick work-around noticed yet other than adding this build support for extensions

### CODE REVIEW

**Comment by YUNQIUGUO** [comment]
- artifact from nuget pipeline: https://artprodcus3.artifacts.visualstudio.com/Abc038106-a83b-4dab-9dd3-5a41bc58f34c/530acbc4-21bc-487d-8cd8-348ff451d2ff/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2FpaW5mcmEvcHJvamVjdElkLzUzMGFjYmM0LTIxYmMtNDg3ZC04Y2Q4LTM0OGZmNDUxZDJmZi9idWlsZElkLzQzMzcwOC9hcnRpZmFjdE5hbWUvb25ueHJ1bnRpbWUtZXh0ZW5zaW9ucy1pb3NfcGFja2FnZQ2/content?format=file&subPath=%2Fpod-archive-onnxruntime-extensions-c-0.11.0-dev%2B433708.245e36f.zip

File: cmake/ext_tests.cmake
Comment (YUNQIUGUO) [inline cmake/ext_tests.cmake:1]: Excluding the C++ tests from build for mac_catalyst only to get around of a linking error happened on pipeline.

File: tools/build.py [hunk 1]
```diff
@@ -204,6 +204,7 @@ def path_from_env_var(env_var: str):
         "--build_apple_framework", action="store_true", help="Build a macOS/iOS framework for ONNX Runtime Extensions."
     )
     parser.add_argument("--ios", action="store_true", help="build for iOS")
+    parser.add_argument("--macos", choices=["MacOSX", "Catalyst"])
```
Comment (skottmckay) [inline tools/build.py:207]: nit: is the casing inconsistent? the `--apple-sysroot` values are all lowercase so wondering if these could/should be the same.

Please add `help=...`.

**Review by skottmckay — APPROVED** [review]

File: nuget/DummyNativeNuget.nuspec [hunk 2]
```diff
@@ -65,5 +65,8 @@
 
     <file src="targets\net6.0-macos\Microsoft.ML.OnnxRuntime.Extensions.Dummy.targets" target="build\net6.0-macos12.3" />
     <file src="targets\net6.0-macos\Microsoft.ML.OnnxRuntime.Extensions.Dummy.targets" target="buildTransitive\net6.0-macos12.3" />
+    <!-- .net 6 macCatalyst -->
+    <file src="targets\net6.0-maccatalyst\_._" target="build\net6.0-maccatalyst14.0" />
+    <file src="targets\net6.0-maccatalyst\_._" target="buildTransitive\net6.0-maccatalyst14.0" />
```
Comment (skottmckay) [inline nuget/DummyNativeNuget.nuspec:70]: If we change the ORT setup to use net7.0 we should do the same here.

**Review by skottmckay — DISMISSED** [review]

File: tools/build.py (hunk 1 above)
Comment (YUNQIUGUO) [reply tools/build.py]: Added a`help=...` message.

yeah the `sysroot` are all in lowercase, however here I was just trying to keep it consistent with ORT options.

https://github.com/microsoft/onnxruntime/blob/e02aef1dedc15e9eed9410b54c5024832e9f3164/tools/ci_build/build.py#L404-L408

**Review by skottmckay — APPROVED** [review]

**Pull request merged**